internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -473,14 +484,41 @@ func (mc *mac) generateRouteKeys(stream *livepeer.CreateStreamResp) []string {
 	return keys
 }
 
+func (mc *mac) sign(sharedSecret string) string {
+	s := sha256.New()
+	s.Write([]byte(sharedSecret))
+	signature := hex.EncodeToString(s.Sum(nil))
+	return signature
+}
+
 func (mc *mac) triggerUserNew(w http.ResponseWriter, r *http.Request, lines []string, rawRequest string) bool {
+
 	if len(lines) != 6 {
 		glog.Errorf("Expected 6 lines, got %d, request \n%s", len(lines), rawRequest)
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write([]byte("false"))
 		return false
 	}
 
+	u := &userNewPayload{
+		StreamName:    	lines[0],
+		IpAddress:     	lines[1],
+		UserAgent:     	lines[2],
+		OutputProtocol: lines[3],
+		RequestUrl:    	lines[4],
+		SessionID:     	lines[5],
+	}
+
+	b, err := json.Marshal(u)
+    if err != nil {
+        glog.Errorf("Error marshalling userNewPayload: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("false"))
```

Comment:
I think we should also allow the request here IMO. An error marshalling means an internal error in our own logic to create the webhook payload, we should avoid breaking playback of all streams if that ever happens (which seems very unlikely anyway)

Reply:
Uncertain I agree; in the event that a user new trigger is configured, the expectation is that a stream is private. Once users start actually using this mechanism, even if there's a failure on our end, I think they'd expect "fail shut" behavior rather than "allow the entire world to watch the stream now".